### PR TITLE
Improve automated tests - add evaluators for different aspects 

### PR DESCRIPTION
**Pull request by katerina-cropguard**
- Migrated to a new langsmith version
- Added evaluators for tools, doc retriever, correctness
- Save experiment results into a csv

### CODE REVIEW

File: croptalk/model_openai_functions.py [hunk 1]
```diff
@@ -101,26 +101,35 @@ def get_model(self) -> Tuple[Chain, AgentTokenBufferMemory]:
...
                 agent=agent,
                 tools=tools,
                 memory=memory,
                 verbose=True,
                 max_iterations=10,
                 return_intermediate_steps=True,
-            )
+            ).with_config(run_name="AgentExecutor")
+        
+        if return_intermediate_steps:
```
Comment (dtourillon-cropguard) [inline croptalk/model_openai_functions.py]: I found it a bit confusing that we use `return_intermediate_steps=True` at L121 (i.e. AgentExecutor's param) regardless of whether `return_intermediate_steps` (i.e. get_model's arg) is True or False... maybe we should add a comment to explain why we have both? And why `return_intermediate_steps=return_intermediate_steps` at L121 is not satisfactory?

File: _scripts/evaluate_overall_performance.py [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+def tool_evaluator(run: Run, example: Example) -> dict:
+    
+    # Accessing the `intermediate_steps`
+    #TODO: Parse the intermediate steps into a list of objects
+    # For now, we just convert the intermediate steps to a string
+    intermediate_steps = run.outputs.get('intermediate_steps')
+    intermediate_steps_str = str(intermediate_steps)
+
+    expected_tool_name = example.outputs.get("Tool")
+    expected_tool = f"tool='{expected_tool_name}'"
```
Comment (dtourillon-cropguard) [inline _scripts/evaluate_overall_performance.py:59]: we're searching for "tool=<tool_name>" in intermediate_steps...
but I see "- tool: FindDocs" in langsmith... which makes me curious: how/why does it work??

image

**Review by dtourillon-cropguard — COMMENTED** [review]
2 questions, otherwise LGTM

File: _scripts/evaluate_overall_performance.py (hunk 2 above)
Comment (simdage) [reply _scripts/evaluate_overall_performance.py:59]: LangSmith returns the tool executed by the agent

**Review by simdage — APPROVED** [review]

File: _scripts/evaluate_overall_performance.py (hunk 2 above)
Comment (dtourillon-cropguard) [reply _scripts/evaluate_overall_performance.py:59]: let's discuss quickly 'cause it's still not clear to me... the test at L60 is if the string expected_tool.lower() in intermediate_steps_str.lower()...

File: croptalk/model_openai_functions.py (hunk 1 above)
Comment (katerina-cropguard) [reply croptalk/model_openai_functions.py]: Updated the naming

**Pull request merged**